extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -130,5 +140,39 @@ void build(CombinedIndexBuildItem index,
                 }
             }
         }
+
+        // manage @BsonProperty for the @ProjectionFor annotation
+        Map<String, String> propertyMapping = new HashMap<>();
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_PROJECTION_FOR)) {
+            Type targetClass = annotationInstance.value().asClass();
+            ClassInfo target = index.getIndex().getClassByName(targetClass.name());
+            for (FieldInfo fieldInfo : target.fields()) {
+                if (fieldInfo.hasAnnotation(DOTNAME_BSON_PROPERTY)) {
+                    AnnotationInstance bsonProperty = fieldInfo.annotation(DOTNAME_BSON_PROPERTY);
+                    propertyMapping.put(fieldInfo.name(), bsonProperty.value().asString());
+                }
+            }
+            for (MethodInfo methodInfo : target.methods()) {
+                if (methodInfo.hasAnnotation(DOTNAME_BSON_PROPERTY)) {
+                    AnnotationInstance bsonProperty = methodInfo.annotation(DOTNAME_BSON_PROPERTY);
+                    propertyMapping.put(methodInfo.name(), bsonProperty.value().asString());
+                }
+            }
+        }
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_PROJECTION_FOR)) {
+            ClassInfo info = annotationInstance.target().asClass();
```

Comment:
I don't think you need to register a transformer per field/method, especially given that each transformer works for the same sets of mappings. Also your transformer can do the transformation for both fields and methods in one go.

Reply:
You're right, the transformer can be reuse for all fields/methods.
I also merge the two transformers. Code is more readable :)